src/main/resources/templates/home.html

```diff
@@ -0,0 +1,41 @@
+<html>
+<body>
+
+<form action="/login" method="post">
+    <input type="text" placeholder="Enter your name" name="userName"/>
+    <button type="submit">Login</button>
+</form>
+
+
+<form action="/add-recipe" method="post">
+    <input type="text" placeholder="Add Recipe" name="createRecipe"/>
+    <button type="submit">Add Recipe</button>
+</form>
+
+
+<ol>
+    {{#recipes}}
+    <li>
+        {{id}} {{text}}
+        <form action="/modify-recipe" method="post">
```

Comment:
This "Edit" form does not have an input to enter the new information.  
Also, the POST action here is called "/modify-recipe", but you have path = "/modify" in your controller!  
Did you test this route?